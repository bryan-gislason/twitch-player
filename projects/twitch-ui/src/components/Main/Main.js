import { useTheme } from "@twitch-player/themes/dist/hooks";
import PropTypes from "prop-types";
import React from "react";
import { StyleSheet, View } from "react-native";
import { useDimensions } from "../../hooks";

const Main = ({ children, testID = "main" }) => {
  const [{ colors }] = useTheme();
  const [dimensions, updateDimensions] = useDimensions("main");

  const rootStyle = [
    styles.root,
    {
      backgroundColor: colors.bodyBackground,
    },
  ];

  return (
    <View
      accessibilityRole="main"
      style={rootStyle}
      testID={testID}
      onLayout={updateDimensions}
    >
      {dimensions && children}
    </View>
  );
};

Main.propTypes = {
  children: PropTypes.node.isRequired,
  testID: PropTypes.string,
};

const styles = StyleSheet.create({
  root: {
    flex: 1,
  },
});

export default Main;
