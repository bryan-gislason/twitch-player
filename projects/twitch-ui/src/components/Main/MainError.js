import { canUseDOM } from "fbjs/lib/ExecutionEnvironment";
import React from "react";
import RetryButton from "../RetryButton";

const reload = () => {
  // istanbul ignore else
  if (canUseDOM) {
    window.location.reload();
  }
};

const MainError = () => {
  return <RetryButton onPress={reload} />;
};

export default MainError;
