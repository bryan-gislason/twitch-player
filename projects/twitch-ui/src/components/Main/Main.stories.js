import { storiesOf } from "@storybook/react";
import React from "react";
import Spinner from "../Spinner";
import Main from ".";

storiesOf("Static Components|Main", module)
  .add("without error", () => {
    return (
      <Main>
        <Spinner />
      </Main>
    );
  })
  .add("with error", () => {
    const BadChild = () => {
      throw new Error("BOOM");
    };
    return (
      <Main>
        <BadChild />
      </Main>
    );
  });
