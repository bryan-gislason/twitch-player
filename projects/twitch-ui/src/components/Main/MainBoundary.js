import React from "react";
import ErrorBoundary from "../ErrorBoundary";
import Main from "./Main";
import MainError from "./MainError";

const MainBoundary = props => {
  return (
    <ErrorBoundary renderError={() => <MainError />}>
      <Main {...props} />
    </ErrorBoundary>
  );
};

export default MainBoundary;
