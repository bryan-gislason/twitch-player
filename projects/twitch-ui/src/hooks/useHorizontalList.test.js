import React from "react";
import { Text } from "react-native";
import { fireEvent, render } from "../testing";
import Main from "../components/Main";
import { useCategoryList, useChannelList } from ".";

describe("useCategoryList", () => {
  const SomeComponent = () => {
    const options = useCategoryList(6);

    return <Text>{`${JSON.stringify(options, null, 2)}`}</Text>;
  };

  let instance;

  beforeEach(() => {
    instance = render(
      <Main>
        <SomeComponent />
      </Main>
    );
  });

  describe("with a small viewport", () => {
    beforeEach(() => {
      fireEvent.resize(window, { target: { width: 480, height: 640 } });
    });

    it("renders with the right attributes", () => {
      expect(instance.getByTestId("main")).toMatchSnapshot();
    });
  });

  describe("with a large viewport", () => {
    beforeEach(() => {
      fireEvent.resize(window, { target: { width: 640, height: 480 } });
    });

    it("renders with the right attributes", () => {
      expect(instance.getByTestId("main")).toMatchSnapshot();
    });
  });
});

describe("useChannelList", () => {
  const SomeComponent = () => {
    const options = useChannelList(6);

    return <Text>{`${JSON.stringify(options, null, 2)}`}</Text>;
  };

  let instance;

  beforeEach(() => {
    instance = render(
      <Main>
        <SomeComponent />
      </Main>
    );
  });

  describe("with a small viewport", () => {
    beforeEach(() => {
      fireEvent.resize(window, { target: { width: 480, height: 640 } });
    });

    it("renders with the right attributes", () => {
      expect(instance.getByTestId("main")).toMatchSnapshot();
    });
  });

  describe("with a large viewport", () => {
    beforeEach(() => {
      fireEvent.resize(window, { target: { width: 640, height: 480 } });
    });

    it("renders with the right attributes", () => {
      expect(instance.getByTestId("main")).toMatchSnapshot();
    });
  });
});
