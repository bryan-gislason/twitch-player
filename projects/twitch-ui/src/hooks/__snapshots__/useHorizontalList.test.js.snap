// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`useCategoryList with a large viewport renders with the right attributes 1`] = `
<main
  class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexGrow-16y2uox rn-flexShrink-1wbh5a2 rn-flexBasis-1ro0kt6 rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  data-testid="main"
  role="main"
  style="background-color: rgb(250, 249, 250);"
>
  <div
    class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-fontFamily-14xgk7a rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
    dir="auto"
  >
    {
  "itemCount": 6,
  "containerWidth": 640,
  "maxImageWidth": 188,
  "maxImageHeight": 250,
  "spaceSides": 30,
  "spaceBetween": 5,
  "spaceBottom": 10,
  "numColumns": 4,
  "numRows": 1,
  "imageWidth": 137,
  "imageHeight": 182
}
  </div>
</main>
`;

exports[`useCategoryList with a small viewport renders with the right attributes 1`] = `
<main
  class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexGrow-16y2uox rn-flexShrink-1wbh5a2 rn-flexBasis-1ro0kt6 rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  data-testid="main"
  role="main"
  style="background-color: rgb(250, 249, 250);"
>
  <div
    class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-fontFamily-14xgk7a rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
    dir="auto"
  >
    {
  "itemCount": 6,
  "containerWidth": 480,
  "maxImageWidth": 188,
  "maxImageHeight": 250,
  "spaceSides": 20,
  "spaceBetween": 5,
  "spaceBottom": 10,
  "numColumns": 3,
  "numRows": 1,
  "imageWidth": 140,
  "imageHeight": 186
}
  </div>
</main>
`;

exports[`useChannelList with a large viewport renders with the right attributes 1`] = `
<main
  class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexGrow-16y2uox rn-flexShrink-1wbh5a2 rn-flexBasis-1ro0kt6 rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  data-testid="main"
  role="main"
  style="background-color: rgb(250, 249, 250);"
>
  <div
    class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-fontFamily-14xgk7a rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
    dir="auto"
  >
    {
  "itemCount": 6,
  "containerWidth": 640,
  "maxImageWidth": 400,
  "maxImageHeight": 225,
  "spaceSides": 30,
  "spaceBetween": 5,
  "spaceBottom": 10,
  "numColumns": 2,
  "numRows": 1,
  "imageWidth": 285,
  "imageHeight": 160
}
  </div>
</main>
`;

exports[`useChannelList with a small viewport renders with the right attributes 1`] = `
<main
  class="rn-alignItems-1oszu61 rn-borderTopStyle-1efd50x rn-borderRightStyle-14skgim rn-borderBottomStyle-rull8r rn-borderLeftStyle-mm0ijv rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-display-6koalj rn-flexGrow-16y2uox rn-flexShrink-1wbh5a2 rn-flexBasis-1ro0kt6 rn-flexDirection-eqz5dr rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-minHeight-ifefl9 rn-minWidth-bcqeeo rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-position-bnwqim rn-zIndex-1lgpqti"
  data-testid="main"
  role="main"
  style="background-color: rgb(250, 249, 250);"
>
  <div
    class="rn-borderTopWidth-13yce4e rn-borderRightWidth-fnigne rn-borderBottomWidth-ndvcnb rn-borderLeftWidth-gxnn5r rn-boxSizing-deolkf rn-color-homxoj rn-display-1471scf rn-fontFamily-14xgk7a rn-fontSize-1b43r93 rn-fontStyle-o11vmf rn-fontVariant-ebii48 rn-fontWeight-gul640 rn-lineHeight-t9a87b rn-marginTop-1mnahxq rn-marginRight-61z16t rn-marginBottom-p1pxzi rn-marginLeft-11wrixw rn-paddingTop-wk8lta rn-paddingRight-9aemit rn-paddingBottom-1mdbw0j rn-paddingLeft-gy4na3 rn-textDecoration-bauka4 rn-whiteSpace-q42fyq rn-wordWrap-qvutc0"
    dir="auto"
  >
    {
  "itemCount": 6,
  "containerWidth": 480,
  "maxImageWidth": 400,
  "maxImageHeight": 225,
  "spaceSides": 20,
  "spaceBetween": 5,
  "spaceBottom": 10,
  "numColumns": 2,
  "numRows": 1,
  "imageWidth": 215,
  "imageHeight": 120
}
  </div>
</main>
`;
